### PR TITLE
    Await for things

### LINKED ISSUE #3085: [BUG]: Some intel compiler flags are greyed out
### Describe the bug

e.g. GCC is currently unable to pick intel asm output

### Steps to reproduce

Visit site, look at GCC. Note it has greyed out intel

### Expected behavior

Intel asm for all!

### Reproduction link

Not applicable

### Screenshots

Not applicable

### Operating System

_No response_

### Browser version

_No response_

### PR DESCRIPTION
**Pull request by mattgodbolt**
We had a number of things that aren't await()ing their promises.
 That led to non-deterministic compiler settings like Intel support.
 I used the IDE to find them all and fix the ones that I thought looked
 important.
  
 More generally we should use tslint (once we're all typescript) to
 ensure this more generally. I tried the eslint equivalent but it
 either overreacted or didn't catch anything.
  
 Closes #3085

### CODE REVIEW

File: static/multifile-service.ts [hunk 1]
```diff
@@ -133,7 +133,7 @@ export class MultifileService {
 
         const zip = await JSZip.loadAsync(f);
 
-        zip.forEach(async (relativePath, zipEntry) => {
+        await Promise.all(zip.map(async (relativePath, zipEntry) => {
```
Comment (partouf) [inline static/multifile-service.ts:136]: Perhaps we should turn this into a normal for loop, it doesn't make sense to use map here, as we never actual return anything. And in that case the return statements can be just continue's

Comment (partouf) [reply static/multifile-service.ts:136]: oh it needs both key and value, hmm

Comment (partouf) [reply static/multifile-service.ts:136]: nevermind then, as long as it works

**Comment by partouf** [comment]
lgtm

**Review by supergrecko — APPROVED** [review]

File: static/multifile-service.ts (hunk 1 above)
Comment (mattgodbolt) [reply static/multifile-service.ts:136]: Up to you: this way the code runs concurrently (as all the promises run together, as they did in the old code). Making it a regular loop would run sequentially (which is cool too, just wasn't sure enough so replaced like with like).

**Pull request merged**